guide/docs/popular-topics/reactions.mdx

```diff
@@ -1,8 +1,321 @@
+# This can be usefull if you want the functionality of buttons, but want a more compact look.
+
+button_emojis = ["✅"]  # What emojis to react to
+reaction_messages = [1060797825417478154]  # What messages to monitor
+
+
+@bot.listen()
+async def on_raw_reaction_add(payload: disnake.RawReactionActionEvent):
+
+    if payload.user_id == bot.user.id:
+        return
+    if not payload.guild_id:
+        return
+    if payload.channel_id not in reaction_messages or str(payload.emoji) not in button_emojis:
+        return
+
+    # Getting the channel, and fetching message as these will be useful
+    event_channel = bot.get_channel(payload.channel_id)
+    event_message = await event_channel.fetch_message(payload.message_id)
+
+    await event_message.remove_reaction(
+        emoji=payload.emoji, member=payload.member
+    )  # Remove the reaction
+    awesome_function()  # Do some stuff
+    await event_channel.send("Done!", delete_after=10.0)
+    # Short message to let the user know it went ok. This is not an interaction so a message response is not strictly needed
+```
+
+</TabItem>
+<TabItem value="reaction_role.py" label="Simple reaction roles">
+
+```python
+# A very simple reaction role system
+
+reaction_messages = [1060797825417478154]  # What messages to monitor
+reaction_roles = {
+    "🎮": 1060778008039919616,
+    "🚀": 1007024363616350308,
+    "<:catpat:967269162386858055>": 1056775021281943583,
+}  # The emojis, and their corresponding role ids
+
+
+@bot.listen()
+async def on_raw_reaction_add(payload: disnake.RawReactionActionEvent):
+
+    # We usually don't want the bot to react to its own actions, nor DM's in this case
+    if payload.user_id == bot.user.id:
+        return
+    if not payload.guild_id:
+        return  # guild_id is None if its a DM
+    if (
+        str(payload.emoji) not in reaction_roles.keys()
+        or payload.message_id not in reaction_messages
+    ):
+        return
+
+    role_to_apply = bot.get_guild(payload.guild_id).get_role(reaction_roles[str(payload.emoji)])
+    if (
+        role_to_apply and not role_to_apply in payload.member.roles
+    ):  # Check if we actually got a role, then check if the member already has it, if not add it
+        await payload.member.add_roles(role_to_apply)
+
+
+@bot.listen()
+async def on_raw_reaction_remove(payload: disnake.RawReactionActionEvent):
+    if payload.user_id == bot.user.id:
+        return
+    if not payload.guild_id:
+        return  # guild_id is None if its a DM
+    if (
+        str(payload.emoji) not in reaction_roles.keys()
+        or payload.message_id not in reaction_messages
+    ):
+        return
+
+    role_to_remove = bot.get_guild(payload.guild_id).get_role(reaction_roles[str(payload.emoji)])
+    if (
+        role_to_apply and role_to_apply in payload.member.roles
+    ):  # Check if we actually got a role, then check if the member actually has it, then remove it
+        await payload.member.remove_roles(role_to_apply)
```

Comment:
```suggestion
@bot.listen()
async def on_raw_reaction_add(payload: disnake.RawReactionActionEvent):
    # We usually don't want the bot to react to its own actions, nor DM's in this case
    if payload.user_id == bot.user.id:
        return
    if not payload.guild_id:
        return  # guild_id is None if its a DM

    role_id = reaction_roles.get(str(payload.emoji))
    if payload.message_id not in reaction_messages or not role_id:
        return

    role = bot.get_guild(payload.guild_id).get_role(role_id)
    # Check if we actually got a role, then check if the member already has it, if not add it
    if role and role not in payload.member.roles:
        await payload.member.add_roles(role)


@bot.listen()
async def on_raw_reaction_remove(payload: disnake.RawReactionActionEvent):
    if payload.user_id == bot.user.id:
        return
    if not payload.guild_id:
        return  # guild_id is None if its a DM

    role_id = reaction_roles.get(str(payload.emoji))
    if payload.message_id not in reaction_messages or not role_id:
        return

    role = bot.get_guild(payload.guild_id).get_role(role_id)
    # Check if we actually got a role, then check if the member actually has it, then remove it
    if role and role in payload.member.roles:
        await payload.member.remove_roles(role)
```

A few more simplifications - once again, Github's diff is useless as ever, proper one [here](https://paste.disnake.dev/?id=1673127644998993)